packages/stylist-core/src/parser.rs

```diff
@@ -20,46 +20,80 @@ use nom::{
 #[cfg(test)]
 use log::trace;
 
-/// A lightweight CSS Parser.
-#[derive(Debug)]
-pub(crate) struct Parser {}
-
+/// Wrap a parser, tracing input and output.
+// if not cfg(test), this would trip up clippy.
 #[allow(clippy::let_and_return)]
-impl Parser {
-    /// Returns Error when string is Empty
-    fn expect_non_empty(i: &str) -> std::result::Result<(), nom::Err<VerboseError<&str>>> {
+fn traced_context<I, O, F>(
```

Comment:
These two functions (`traced_context` and `expect_non_empty`) should be within the impl block of `Parser`?

Reply:
I moved them outside, cause unlike `Parser::trimmed`, they are general combinators and don't depend on any css parser specifics. Thought that gave a better impression of that fact, nothing wrong with freestanding methods, or am I wrong?

Reply:
Well, I guess it's OK to have it outside of the `Parser`? Especially given that we may create different parsers for different purposes (e.g.:`keyframes`)